socket.go

```diff
@@ -70,8 +70,28 @@ func (s *SocketServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			}
 		}
 	}
+	clientID := r.URL.Query().Get("client-id")
```

Comment:
JS websocket API does not allow for setting custom headers so used query params instead.

Reply:
Other option would be to send the id on the first message or something but I think this is better

Reply:
Yep, I think as well. No need to clutter our custom protocol even more.